src/Misc/MiddleWare.cpp

```diff
@@ -298,14 +298,49 @@ struct NonRtObjStore
         return objmap[loc];
     }
 
-    void handleOscil(const char *msg, rtosc::RtData &d) {
-        string obj_rl(d.message, msg);
+    //! try to dispatch a message at the OscilGen non-RT-ports
+    void handleOscilADnote(const char *msg, rtosc::RtData &d, bool voiceOscil) {
+        const string obj_rl(d.message, msg);
         void *osc = get(obj_rl);
         if(osc)
         {
             strcpy(d.loc, obj_rl.c_str());
             d.obj = osc;
             OscilGen::non_realtime_ports.dispatch(msg, d);
+            if(d.matches == 0 /* || d.forwarded -> check should be useless, can't be forwarded */)
```

Comment:
Seems good enough to start, though additional changes can be used to avoid unneeded recomputation (e.g. reading parameters should not make it look like things have changed).

Reply:
I now added as a condition that the wavetable is only regenerated if the OSC message had any arguments. This seems to work good currently.